ros2/source/source/Dockerfile

```diff
@@ -73,26 +78,37 @@ RUN pip3 install -U \
     flake8-quotes \
     pytest \
     pytest-cov \
-    pytest-runner
+    pytest-repeat \
+    pytest-rerunfailures \
+    pytest-runner \
+    vcstool
+
+# clone and build buildtool
+ENV COLCON_WS /root/ros2/colcon
+RUN mkdir -p $COLCON_WS/src
+WORKDIR $COLCON_WS
+RUN wget https://raw.githubusercontent.com/colcon/colcon.readthedocs.org/master/colcon.repos && \
+    vcs import src < colcon.repos
+RUN ./src/colcon-core/bin/colcon build --paths src/*
+RUN . install/prefix.sh && colcon build
 
 # clone source
-ENV ROS2_WS /root/ros2_ws
+ENV ROS2_WS /root/ros2/ros2_ws
 RUN mkdir -p $ROS2_WS/src
 WORKDIR $ROS2_WS
-RUN wget https://raw.githubusercontent.com/ros2/ros2/release-latest/ros2.repos \
+RUN wget https://raw.githubusercontent.com/ros2/ros2/master/ros2.repos \
     && vcs import src < ros2.repos
 
 # build source
 WORKDIR $ROS2_WS
-RUN src/ament/ament_tools/scripts/ament.py \
+RUN . $COLCON_WS/install/prefix.sh && colcon \
     build \
-    --build-tests \
-    --cmake-args -DSECURITY=ON -- \
-    --isolated \
-    --parallel \
-    --symlink-install
+    --cmake-args " -DSECURITY=ON" " -DBUILD_TESTING=ON" \
+    --symlink-install \
+    --event-handler console_cohesion+
 
 # setup bashrc
+ENV ROS2_WS /root/ros2/ros2_ws
```

Comment:
This is redundant given line 96

Reply:
You're totally right.
Sorry for wasting your review time, this is currently playground and far from being ready for review. I just pushed a rebased version because someone else needed to do some debugging on bionic.

I will open a PR once this branch is closer to a reviewable state

Reply:
If you prefer I can do my playground on a fork in the future to avoid spurious email notifications ?